controllers/updateservice_controller.go

```diff
@@ -350,6 +351,38 @@ func (r *UpdateServiceReconciler) ensureTrustedClusterCA(ctx context.Context, re
 	return err
 }
 
+func (r *UpdateServiceReconciler) ensureGraphDataSHA(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService,
+	resources *kubeResources) (string, error){
+
+	pod := resources.graphDataPodContainer
+	if err := controllerutil.SetControllerReference(instance, pod, r.Scheme); err != nil {
+		return "", err
+	}
+
+	// Check if this pod already exists
+	found := &corev1.Pod{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+
+	if err != nil && apiErrors.IsNotFound(err) || found.Status.Phase == "Succeeded"{
+		reqLogger.Info("Creating Pod", "Namespace", pod.Namespace, "Name", pod.Name)
+		err := r.Client.Create(ctx, pod)
```

Comment:
Probably need a `Delete` in the `Succeeded` case.  And also maybe a delete in the "we used  to be a by-tag reference, but now the graph-data pullspec is a by-digest reference, so [we don't need to bother launching Pods anymore][1]" case?

[1]: https://github.com/openshift/cincinnati-operator/pull/164#discussion_r1119454227

Reply:
added a delete on success condition

Reply:
Is there a chance of racing with someone else (is it possible that we run multiple replicas?) creating the pod? It is possible that we Get, see nothing, something else creates the pod, we create and fail with conflict... I'm fine if we say that that cannot happen realistically.

Reply:
> because, we're creating a pod, it can only have 1 pod running at a time with the given name

Petr was pointing out that other actors (e.g. a runaway second cincinnati-operator replica) might also create a pod with the same name.  And actually, because `ensureGraphDataSHA` is checking a single UpdateService, it seems like an operator managing several by-tag UpdateService would probably struggle like:

1. Reconciling UpdateService A, create a `graph-data-tag-digest` pod.
2. Reconciling UpdateService B, pod is still running, return `"", nil`.
3. Reconciling UpdateService A, pod is still running, return `"", nil`.
4. Reconciling UpdateService B, pod is `Succeeded`, return the image ID.  But this is bad, because that image ID  was for the by-tag pullspec from UpdateService A, not the pullspec from B.

Instead, you'll want to somehow uniquify by pullspec-under test.  Perhaps `SetControllerReference` allows you to clearly associate a pod with the appropriate UpdateService.  And then Kube's built-in garbage collection will handle deleting any pods whose UpdateService is deleted.